### PR TITLE
windows drivers' cases: Update windows driver status check function

### PR DESCRIPTION
**Pull request by lijinlijin**
msinfo32 has performance issues, windrv_check_running_verifier() is re-worked with devcon tool. So update related code in cases.

ID：1598291

Signed-off-by: Li Jin <lijin@redhat.com>

### CODE REVIEW

**Comment by lijinlijin** [comment]
Depends on
https://gitlab.cee.redhat.com/kvm-qe/kar/merge_requests/103
https://github.com/avocado-framework/avocado-vt/pull/1892

**Comment by lijinlijin** [comment]
(01/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.balloon_stress: PASS (4629.74 s)
 (02/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.driver_load_stress.with_viostor.before_bg_test: PASS (151.85 s)
 (03/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.driver_load_stress.with_vioscsi.before_bg_test: PASS (160.58 s)
 (04/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.driver_load_stress.with_netkvm.before_bg_test: PASS (308.14 s)
 (05/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.driver_load_stress.with_balloon.before_bg_test: PASS (346.80 s)
 (06/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.block_resize: PASS (72.31 s)
 (07/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.block_resize.data_plane: PASS (74.39 s)
 (08/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.netkvm_in_use.with_stop_continue.before_bg_test: PASS (85.92 s)
 (09/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.ctrl_vlan.ctrl_on: PASS (49.51 s)
 (10/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.ctrl_vlan.ctrl_off: PASS (34.19 s)
 (11/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.balloon_service.small_polling_interval: PASS (837.52 s)
 (12/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.balloon_service.sc_interrogate: PASS (226.24 s)
 (13/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.pvpanic.with_stop_continue: PASS (425.04 s)
 (14/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.single_driver_install.with_netkvm.virtio_win_iso_media: PASS (100.47 s)
 (15/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.single_driver_install.with_viorng.virtio_win_iso_media: PASS (63.53 s)
 (16/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.single_driver_install.with_viostor.virtio_win_iso_media: PASS (33.99 s)
 (17/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.single_driver_install.with_vioscsi.virtio_win_iso_media: PASS (61.27 s)
 (18/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.single_driver_install.with_vioserial.virtio_win_iso_media: PASS (61.85 s)
 (19/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.single_driver_install.with_balloon.virtio_win_iso_media: PASS (62.02 s)
 (20/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.single_driver_install.with_pvpanic.virtio_win_iso_media: PASS (61.11 s)
 (21/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.single_driver_install.with_vioinput.device_mouse.virtio_win_iso_media: PASS (61.19 s)
 (22/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.single_driver_install.with_vioinput.device_keyboard.virtio_win_iso_media: PASS (62.32 s)
 (23/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.single_driver_install.with_vioinput.device_tablet.virtio_win_iso_media: PASS (61.90 s)
 (24/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.balloon_in_use.with_stop_continue.before_bg_test: PASS (347.59 s)
 (25/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.pvpanic_basic.nmi: PASS (247.81 s)
 (26/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.balloon_hotplug: PASS (603.70 s)
 (27/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.win_irq_check.with_balloon: PASS (32.81 s)
 (28/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.win_irq_check.with_viorng: PASS (55.87 s)
 (29/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.win_sigverif.with_netkvm: PASS (67.31 s)
 (30/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.win_sigverif.with_viorng: PASS (67.50 s)
 (31/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.win_sigverif.with_viostor: PASS (68.11 s)
 (32/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.win_sigverif.with_vioscsi: PASS (67.49 s)
 (33/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.win_sigverif.with_vioserial: PASS (68.07 s)
 (34/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.win_sigverif.with_balloon: PASS (67.63 s)
 (35/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.win_sigverif.with_pvpanic: PASS (67.93 s)
 (36/36) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.win_sigverif.with_vioinput: PASS (66.94 s)

File: generic/tests/iozone_windows.py [hunk 1]
```diff
@@ -63,9 +63,14 @@ def get_driver():
     session = vm.wait_for_login(timeout=timeout)
     driver_name = get_driver()
     if driver_name:
-        session = utils_test.qemu.windrv_check_running_verifier(session, vm,
-                                                                test, driver_name,
+        driver_id_pattern = params["driver_id_pattern"]
+        devcon_path = utils_misc.set_winutils_letter(session, params["devcon_path"])
```
Comment (hereischen) [inline generic/tests/iozone_windows.py]: `devcon_path = "WIN_UTILS:\devcon\x86\"` probably needs to added in upstream as well, or any one who run the script will be failed.

Comment (lijinlijin) [reply generic/tests/iozone_windows.py]: Updated it in upstream guest-os file in https://github.com/avocado-framework/avocado-vt/pull/1892
Thank you for the advice.

Comment (xiagao) [inline generic/tests/iozone_windows.py]: How about move utils_misc.set_winutils_letter to utils_test.qemu.windrv_check_running_verifier() function, which will save codes.

**Comment by lijinlijin** [comment]
@luckyh 
Modified according to your suggestions, thanks

**Comment by lijinlijin** [comment]
latest result:
 (1/3) Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.rng_bat: PASS (31.42 s)
 (2/3) Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.win_irq_check.with_balloon: PASS (31.66 s)
 (3/3) Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.win_irq_check.with_viorng: PASS (31.65 s)

**Comment by lijinlijin** [comment]
@hereischen @xiagao 
Any further comment?
Thanks

File: provider/win_dev.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+"""
+import re
+
+from virttest import error_context
+from virttest import utils_misc
+
+
+@error_context.context_aware
+def get_hwid(session, test, hwid_pattern, devcon_path, timeout=300):
```
Comment (luckyh) [inline provider/win_dev.py]: Why not keep `hwid_pattern` in this function?

File: provider/win_dev.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+"""
+import re
+
+from virttest import error_context
+from virttest import utils_misc
+
+
+@error_context.context_aware
+def get_hwid(session, device_name, devcon_path, timeout=300):
```
Comment (luckyh) [inline provider/win_dev.py]: Should be `get_hwids()`?

**Comment by lijinlijin** [comment]
depends on https://github.com/avocado-framework/avocado-vt/pull/1925

**Comment by lijinlijin** [comment]
(1/2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.win_irq_check.with_balloon: PASS (31.79 s)
 (2/2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.win_irq_check.with_viorng: PASS (31.62 s)

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**